### PR TITLE
chore(workflow): setup project

### LINKED ISSUE #1: setup project
Set up project with:

- [x] [eslint](https://eslint.org/)
- [x] [pretttier](https://prettier.io/)
- [x] [react](https://reactjs.org/)
- [x] [typescript](https://www.typescriptlang.org/)
- [x] [react router](https://reactrouter.com/en/main)
- [x] [tanstack-query](https://tanstack.com/query/latest)
- [x] [vite](https://vitejs.dev/)
- [x] [firebase](https://firebase.google.com/)
- [x] [react material ui](https://mui.com/)
- [x] [react-hook-form](https://react-hook-form.com/)
- [x] [git-hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks)
- [x] [i18n](https://react.i18next.com/)
- [x] [google maps api](https://developers.google.com/maps/documentation/javascript/overview)

### PR DESCRIPTION
**Pull request by kaczor6418**
Under following url you can find documentation of project structure:
https://github.com/GDSC-Lodz-University-of-Technology/warzywniax/wiki/Project-structure

closes #1

### CODE REVIEW

File: src/main.tsx [hunk 1]
```diff
@@ -1,10 +1,16 @@
-import React from 'react'
-import ReactDOM from 'react-dom/client'
-import App from './App'
-import './index.css'
+import './locales/i18n';
+import './index.css';
+import * as React from 'react';
+import * as ReactDOM from 'react-dom/client';
+import { CUSTOM_THEME } from './configs/mui.config';
+import { ROUTER } from './router/router';
```
Comment (HKGx) [inline src/main.tsx:6]: [*]

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    "no-await-in-loop": "off",
+    "no-shadow": "off",
+    "no-constant-binary-expression": "off",
+    "no-template-curly-in-string": "off",
+    "no-unmodified-loop-condition": "off",
+    "no-use-before-define": "off",
+    "sort-keys": "off",
+    "accessor-pairs": "off",
+    "no-magic-numbers": "off",
+    "@typescript-eslint/no-magic-numbers": "off",
```
Comment (HKGx) [inline .eslintrc.json:188]: magic numbers make it harder to comprehend code!!!

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    "@typescript-eslint/no-namespace": "off",
```
Comment (HKGx) [inline .eslintrc.json:189]: namespaces are pretty much outdated, we shouldn't disable this

File: .eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    "react/react-in-jsx-scope": "off"
```
Comment (HKGx) [inline .eslintrc.json:190]: [no-constant-binary-expression](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/react-in-jsx-scope.md#when-not-to-use-it)

> If you are using the [new JSX transform from React 17](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports), you should disable this rule by extending [react/jsx-runtime](https://github.com/jsx-eslint/eslint-plugin-react/blob/8cf47a8ac2242ee00ea36eac4b6ae51956ba4411/index.js#L165-L179) in your eslint config (add "plugin:react/jsx-runtime" to "extends").

File: .eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+      }
+    ],
+    //Disabled
+    "no-await-in-loop": "off",
+    "no-shadow": "off",
+    "no-constant-binary-expression": "off",
+    "no-template-curly-in-string": "off",
+    "no-unmodified-loop-condition": "off",
+    "no-use-before-define": "off",
+    "sort-keys": "off",
```
Comment (HKGx) [inline .eslintrc.json:185]: I like sorted keys but whatever

File: .eslintrc.json [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+        "destructuredArrayIgnorePattern": "^_"
+      }
+    ],
+    //Disabled
+    "no-await-in-loop": "off",
+    "no-shadow": "off",
+    "no-constant-binary-expression": "off",
+    "no-template-curly-in-string": "off",
+    "no-unmodified-loop-condition": "off",
+    "no-use-before-define": "off",
```
Comment (HKGx) [inline .eslintrc.json:184]: what's wrong with this rule?

**Review by HKGx — CHANGES_REQUESTED** [review]
I commented mostly on ESLint rules that I think are unnecessary/bad.
I believe we should use https://github.com/trivago/prettier-plugin-sort-imports as it makes handling imports easier, but it can be added later 👍

File: .prettierrc.json [hunk 7]
```diff
@@ -0,0 +1,18 @@
+{
+  "printWidth": 150,
```
[the remark this answers is not in the document]
Comment (kaczor6418) [reply .prettierrc.json:2]: I think for most modern screens this should be okay

File: src/router/router.tsx [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+import { App } from '../App';
+import { offersRoute } from './offers.route';
+import { shopsRoute } from './shops.route';
+
+const rootRoute: RouteObject = {
+  element: <App />,
+  path: '/',
+};
+
+export const ROUTER = createBrowserRouter([rootRoute, offersRoute, shopsRoute]);
```
[the remark this answers is not in the document]
Comment (kaczor6418) [reply src/router/router.tsx:11]: `router`

File: .prettierrc.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (kaczor6418) [reply .prettierrc.json:2]: `100`

File: src/configs/mui.config.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    md: false;
+    lg: false;
+    xl: false;
+    mobile: true;
+    tablet: true;
+    desktop: true;
+  }
+}
+
+export const CUSTOM_THEME = createTheme({
```
[the remark this answers is not in the document]
Comment (kaczor6418) [reply src/configs/mui.config.ts:16]: camelCase notation

File: src/configs/firebase.config.ts [hunk 10]
```diff
@@ -0,0 +1,5 @@
+import { initializeApp } from 'firebase/app';
+
+export const FIREBASE_CONNECTION = initializeApp({
```
[the remark this answers is not in the document]
Comment (kaczor6418) [reply src/configs/firebase.config.ts:3]: camelCase notation

File: src/router/offers.route.tsx [hunk 11]
```diff
@@ -0,0 +1,7 @@
+import { OffersPage } from '../pages/OffersPage/OffersPage';
+import { RouteObject } from 'react-router-dom';
+
+export const offersRoute: RouteObject = {
```
[the remark this answers is not in the document]
Comment (kaczor6418) [reply src/router/offers.route.tsx:4]: camelCase notation

File: src/router/shops.route.tsx [hunk 12]
```diff
@@ -0,0 +1,7 @@
+import { RouteObject } from 'react-router-dom';
+import { ShopsPage } from '../pages/ShopsPage/ShopsPage';
+
+export const shopsRoute: RouteObject = {
```
[the remark this answers is not in the document]
Comment (kaczor6418) [reply src/router/shops.route.tsx:4]: camelCase notation

File: .eslintrc.json [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+    "require-yield": "error",
+    "sort-imports": [
+      "error",
+      {
+        "ignoreCase": true
+      }
+    ],
+    "yoda": ["error", "never"],
+    "prettier/prettier": "error",
+    "import/no-default-export": "error",
```
[the remark this answers is not in the document]
Comment (kaczor6418) [reply .eslintrc.json:154]: let's keep this

File: src/main.tsx (hunk 1 above)
Comment (kaczor6418) [reply src/main.tsx:6]: camelCase notation

File: .eslintrc.json [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+    "no-unexpected-multiline": "error",
+    "no-unreachable": "error",
+    "no-unreachable-loop": "error",
+    "no-unsafe-finally": "error",
+    "no-unsafe-negation": "error",
+    "no-unsafe-optional-chaining": "error",
+    "no-useless-backreference": "error",
+    "use-isnan": "error",
+    "valid-typeof": "error",
+    "arrow-body-style": ["error", "always"],
```
[the remark this answers is not in the document]
Comment (kaczor6418) [reply .eslintrc.json:72]: as-needed

File: .eslintrc.json [hunk 15]
```diff
@@ -0,0 +1,193 @@
...
+    "no-useless-concat": "error",
+    "no-useless-constructor": "error",
+    "no-useless-escape": "error",
+    "no-useless-rename": "error",
+    "no-useless-return": "error",
+    "no-var": "error",
+    "no-with": "error",
+    "object-shorthand": ["error", "consistent-as-needed"],
+    "operator-assignment": ["error", "always"],
+    "prefer-arrow-callback": "error",
```
[the remark this answers is not in the document]
Comment (kaczor6418) [reply .eslintrc.json:137]: this can stay

File: .eslintrc.json [hunk 16]
```diff
@@ -0,0 +1,193 @@
...
+    ],
+    "@typescript-eslint/no-unused-vars": [
+      "error",
+      {
+        "argsIgnorePattern": "^_",
+        "destructuredArrayIgnorePattern": "^_"
+      }
+    ],
+    //Disabled
+    "no-await-in-loop": "off",
```
[the remark this answers is not in the document]
Comment (kaczor6418) [reply .eslintrc.json:179]: let's turn it on

File: .eslintrc.json (hunk 2 above)
Comment (kaczor6418) [reply .eslintrc.json:188]: so lets keeps this as it is

File: .eslintrc.json (hunk 3 above)
Comment (kaczor6418) [reply .eslintrc.json:189]: turn it on

File: .eslintrc.json (hunk 4 above)
Comment (kaczor6418) [reply .eslintrc.json:190]: ok

File: .eslintrc.json (hunk 5 above)
Comment (kaczor6418) [reply .eslintrc.json:185]: turn it on

File: .eslintrc.json (hunk 6 above)
Comment (kaczor6418) [reply .eslintrc.json:184]: we are using typescript so this is not needed as TS doesn't allow use variable before declaration

File: src/locales/i18n.ts [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  pl: {
+    components: {
+      shops: shopsPL,
+      offers: offersPL,
+    },
+  },
+};
+
+void i18n.use(initReactI18next).init({
```
[the remark this answers is not in the document]
Comment (kaczor6418) [reply src/locales/i18n.ts:23]: to resolve unhandled promise (hack to satisfy TypeScript)

**Review by mkidawa — APPROVED** [review]
LGTM

**Review by HKGx — APPROVED** [review]
:+1:

**Pull request merged**